qiskit/aqua/algorithms/factorizers/shor.py

```diff
@@ -274,21 +275,21 @@ def egcd(a: int, b: int) -> Tuple[int, int, int]:
                              "modular inverse does not exist.".format(a, m, g))
         return x % m
 
-    def _get_factors(self, output_desired: str, t_upper: int) -> bool:
+    def _get_factors(self, measurement: str) -> Union[None, List[int]]:
```

Comment:
`Optional[...]` is a alias for `Union[None, ...]`
```suggestion
    def _get_factors(self, measurement: str) -> Optional[List[int]]:
```